modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/extension/discovery/ExtensionDiscoveryRequest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.api.extension.discovery;
+
+import static java.util.Collections.emptySet;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.extension.api.loader.DeclarationEnricher;
+import org.mule.runtime.module.artifact.activation.internal.extension.discovery.DefaultExtensionDiscoveryRequest;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactPluginDescriptor;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Container of the parameters that can be given to
+ * {@link ExtensionModelDiscoverer#discoverPluginsExtensionModels(ExtensionDiscoveryRequest)}.
+ * 
+ * @since 4.5
+ */
+public interface ExtensionDiscoveryRequest {
+
+  /**
+   * @return a fluent builder for creating a new {@link ExtensionDiscoveryRequest} instance.
+   */
+  static ExtensionDiscoveryRequestBuilder builder() {
+    return new ExtensionDiscoveryRequestBuilder();
+  }
+
+  /**
+   * @return {@link ArtifactPluginDescriptor}s for artifact plugins deployed inside the artifact. Non-null.
+   */
+  List<ArtifactPluginDescriptor> getArtifactPlugins();
```

Comment:
rename to make clear that this reutrns the descriptors and not the actual plugins

Reply:
Done.